addons/mrp/models/mrp_bom.py

```diff
@@ -104,6 +106,11 @@ def _compute_possible_product_template_attribute_value_ids(self):
         for bom in self:
             bom.possible_product_template_attribute_value_ids = bom.product_tmpl_id.valid_product_template_attribute_line_ids._without_no_variant_attributes().product_template_value_ids._only_active()
 
+    @api.depends('company_id')
+    def _compute_currency_id(self):
+        for company in self:
+            company.currency_id = company.company_id.sudo().currency_id.id or self.env.company.currency_id.id
```

Comment:
you need a sudo on first part but not on the second part but it's the same object

Reply:
Hello @amoyaux,

The first is a production object (production.company_id), and the second is an environment object (self.env.company).

Thanks.